src/main/java/hu/iac/webshop/controllers/OrderProductController.java

```diff
@@ -29,19 +28,38 @@ public OrderProductController(OrderProductService orderProductService, OrderServ
     }
 
     @PostMapping("/orderproducts")
-    public ResponseEntity<OrderProduct> create(@Valid @RequestBody OrderProductRequest orderProductRequest){
+    public ResponseEntity create(@Valid @RequestBody OrderProductRequest orderProductRequest){
         Optional<Order> order = this.orderService.find(orderProductRequest.getOrderId());
         Optional<Product> product = this.productService.find(orderProductRequest.getProductId());
 
         if (order.isPresent() && product.isPresent()) {
-            OrderProduct orderProduct = new OrderProduct(
-                order.get(),
-                product.get(),
-                orderProductRequest.getAmount()
-            );
-            return new ResponseEntity<OrderProduct>(this.orderProductService.create(orderProduct), HttpStatus.OK);
+            int amount = orderProductRequest.getAmount();
+            int stock = product.get().getStock();
+
+            if (amount <= stock) {
+                OrderProduct orderProduct = new OrderProduct(
+                    order.get(),
+                    product.get(),
+                    orderProductRequest.getAmount()
+                );
+                return new ResponseEntity<OrderProduct>(this.orderProductService.create(orderProduct), HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>("Not enough products in stock to order this amount", HttpStatus.NOT_ACCEPTABLE);
+            }
         }
 
         return new ResponseEntity<>(HttpStatus.NOT_FOUND);
     }
+
+    @DeleteMapping("/orderproducts/{orderId}/{productId}")
```

Comment:
is /order/{orderId}/product/{productId} niet netter?
weet dat dit hierboven ook zo staat hoor dus kijk maar effe

Reply:
Ja, stukken netter, guud idee